src/compiler/program.ts

```diff
@@ -313,7 +313,7 @@ module ts {
             }
             let diagnostic: DiagnosticMessage;
             if (hasExtension(fileName)) {
-                if (!options.allowNonTsExtensions && !fileExtensionIs(host.getCanonicalFileName(fileName), ".ts")) {
+                if (!options.allowNonTsExtensions && !forEach(supportedExtensions, extension => fileExtensionIs(host.getCanonicalFileName(fileName), extension))) {
                     diagnostic = Diagnostics.File_0_must_have_extension_ts_or_d_ts;

```

Comment:
this diagnostic is now a lie.  We shoudl print: "The only supported extensions are `{0}`" and then list the extensions in that array.